adapters/richaudience/richaudience.go

```diff
@@ -0,0 +1,355 @@
+package richaudience
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/mxmCherry/openrtb/v15/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/metrics"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+type richaudienceRequest struct {
+	ID     string             `json:"id,omitempty"`
+	Imp    []openrtb2.Imp     `json:"imp,omitempty"`
+	User   richaudienceUser   `json:"user,omitempty"`
+	Device richaudienceDevice `json:"device,omitempty"`
+	Site   richaudienceSite   `json:"site,omitempty"`
+	Test   int8               `json:"test,omitempty"`
+}
+
+type richaudienceUser struct {
+	BuyerUID string              `json:"buyeruid,omitempty"`
+	Ext      richaudienceUserExt `json:"ext,omitempty"`
+}
+
+type richaudienceUserExt struct {
+	Eids    []openrtb_ext.ExtUserEid `json:"eids,omitempty"`
+	Consent string                   `json:"consent,omitempty"`
+}
+
+type richaudienceDevice struct {
+	//GEO only Java, IFA
```

Comment:
What is the purpose of this comment? I don't understand what you're trying to convey. Is this a dev note you didn't remove? I see there are many more comments like this one.

Reply:
in next commit i have solved